src/okta/fetchBatchOfApplicationUsers.ts

```diff
@@ -34,7 +34,11 @@ export default async function fetchBatchOfApplicationUsers(
 
   const applicationsState = await applicationCache.getState();
   if (!applicationsState || !applicationsState.fetchCompleted) {
-    throw new IntegrationError("Application fetching did not complete");
+    throw new IntegrationError({
+      message:
+        "Error in step `fetch-application-users`: Application fetching did not complete. This may occur when the `fetch-applications` step dependency fails.",
```

Comment:
I think it good that we give a bit more detail to the user by marking it `expose: true` and improving the message.

Since we know for a fact that `fetch-applications` failed, what do you think about:
```suggestion
        "Step dependency failed, cannot ingest application users: 'fetch-applications'",
```
I cannot recall or easily verify ATM whether the current step name needs to be repeated in the message.